src/openqaoa-core/optimizers/qaoa_optimizer.py

```diff
@@ -0,0 +1,80 @@
+from .training_vqa import (
+    ScipyOptimizer,
+    CustomScipyGradientOptimizer,
+    PennyLaneOptimizer,
+)
+from ..qaoa_components import QAOAVariationalBaseParams
+from ..backends.basebackend import VQABaseBackend
+
+
+def available_optimizers():
+    """
+    Return a list of available optimizers.
+    """
+
+    optimizers = {
+        "scipy": ScipyOptimizer.SCIPY_METHODS,
+        "custom_scipy_gradient": CustomScipyGradientOptimizer.CUSTOM_GRADIENT_OPTIMIZERS,
+        "custom_scipy_pennylane": PennyLaneOptimizer.PENNYLANE_OPTIMIZERS,
+    }
+
+    return optimizers
+
+
+def get_optimizer(
+    vqa_object: VQABaseBackend,
+    variational_params: QAOAVariationalBaseParams,
+    optimizer_dict: dict,
+):
+    """
+    Initialise the specified optimizer class with provided method and
+    optimizer-specific options
+
+    Parameters
+    ----------
+    vqa_object:
+        Backend object of class VQABaseBackend which contains information on the
+        backend used to perform computations, and the VQA circuit.
+
+    variational_params:
+        Object of class QAOAVariationalBaseParams, which contains
+        information on the circuit to be executed,
+        the type of parametrisation, and the angles of the VQA circuit.
+
+    optimizer_dict:
+        Optimizer information dictionary used to construct
+        the optimizer with specified options
+
+    Returns
+    -------
+    optimizer:
+        Optimizer object of type specified by specified method
+    """
+    SUPPORTED_OPTIMIZERS = {
+        "scipy": ScipyOptimizer,
+        "custom_scipy_gradient": CustomScipyGradientOptimizer,
+        "custom_scipy_pennylane": PennyLaneOptimizer,
+    }
+
+    method = optimizer_dict["method"].lower()
+    optimizers = available_optimizers()
+
+    method_valid = False
+    for opt_class, methods in optimizers.items():
+        if method in methods:
+            selected_class = opt_class
+            method_valid = True
+
+    assert method_valid, ValueError(
+        f"Selected optimizer method '{method}' is not supported."
+        f"Please choose from {available_optimizers()}"
+    )
+
+    try:
```

Comment:
why do we have a try and exception if we raise all exceptions?

Reply:
Good point, ill remove it and see if its causing any problem.

Reply:
Resolved in commit [a52535a](https://github.com/entropicalabs/openqaoa/pull/170/commits/a52535a3691816acfd4139b69ba58105dfa01c35)